File: updated_code/code/java/145-binary-tree-postorder-traversal.java
package com.example;

import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        // Initialize the result list
        List<Integer> result = new ArrayList<Integer>();

        // Initialize the stack
        Stack<TreeNode> stack = new Stack<TreeNode>();

        // Push the root node onto the stack
        if (root != null) {
            stack.push(root);
        }

        // Traverse the binary tree using a stack
        while (!stack.isEmpty()) {
            // Pop the top node from the stack
            TreeNode node = stack.pop();

            // Add the node's value to the result list
            result.add(node.val);

            // Push the node's left and right children onto the stack
            if (node.right != null) {
                stack.push(node.right);
            }
            if (node.left != null) {
                stack.push(node.left);
            }
        }

        // Reverse the result list to get the postorder traversal
        Collections.reverse(result);

        return result;
    }
}

File: updated_code/code/python/145-binary-tree-postorder-traversal.py
from typing import Optional, List
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    """
    This class solves the postorder traversal of a binary tree.

    Args:
    root (Optional[TreeNode]): The root of the binary tree.

    Returns:
    List[int]: The postorder traversal of the binary tree.
    """
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        """
        This method performs the postorder traversal of the binary tree.

        Args:
        root (Optional[TreeNode]): The root of the binary tree.

        Returns:
        List[int]: The postorder traversal of the binary tree.
        """
        if root is None:
            return []

        result = []
        stack = [root]

        while stack:
            node = stack.pop()
            result.append(node.val)

            if node.left:
                stack.append(node.left)
            if node.right:
                stack.append(node.right)

        return result[::-1]